substratevm/src/com.oracle.svm.test.jdk11/src/com/oracle/svm/test/jdk11/jfr/utils/JFRFileParser.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, 2021, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.test.jdk11.jfr.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Set;
+
+import com.oracle.svm.jfr.JfrChunkWriter;
+import com.oracle.svm.jfr.JfrTypes;
+
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ClassConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ClassLoaderConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.FrameTypeConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.MethodConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ModuleConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.NotFoundConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.PackageConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.StacktraceConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.SymbolConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ThreadConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ThreadGroupConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.poolparsers.ThreadStateConstantPoolParser;
+import jdk.jfr.Recording;
+
+public class JFRFileParser {
+
+    private static class Positions {
```

Comment:
I tend to place private static classes at the end of the file as that code is usually not the code that someone wants to see when opening this class.

Reply:
Make sense.